File: Engine/src/jvmMain/java/org/etieskrill/engine/input/controller/KeyCameraController.java
package org.etieskrill.engine.input.controller;

import org.etieskrill.engine.graphics.camera.Camera;

public class KeyCameraController extends KeyCharacterController<Camera> {

    public KeyCameraController(Camera target) {
        super(target, (delta, camera, deltaPosition, speed) -> {
            deltaPosition.x *= -1;
            camera.translate(deltaPosition.mul((float) (delta * speed)));
        });
    }

}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/input/controller/KeyCameraController.kt
package org.etieskrill.engine.input.controller

import org.etieskrill.engine.graphics.camera.Camera
import org.joml.times

class KeyCameraController(
    camera: Camera
) : KeyCharacterController<Camera>(camera, { delta, camera, deltaPosition, speed ->
    deltaPosition.x = -deltaPosition.x
    camera.translate(deltaPosition * delta.toFloat() * speed)
})
